aea/aea_builder.py

```diff
@@ -1334,13 +1346,25 @@ def build(
         self._build_called = True
         return aea
 
-    def get_default_ledger(self) -> str:
+    def get_default_ledger(self) -> Optional[str]:
         """
         Return default ledger.
 
+        :return: the default ledger identifier, or None.
+        """
+        return self._default_ledger
+
+    def get_default_ledger_not_null(self) -> str:
+        """
+        Return default ledger if not None, else raise error.
+
         :return: the default ledger identifier.
+        :raise: Value
```

Comment:
```suggestion
        :raise: ValueError when the default ledger field is None.
```

Reply:
See below, nullable makes everything more complicated for no good reason.